### PR TITLE
sort plugin

### PR DESCRIPTION
**Pull request by YangfanZhang**
Sort plugin by names, search plugin, sort plugin state key

### CODE REVIEW

**Review by YangfanZhang — COMMENTED** [review]
Added sort plugin names, search plugin by names, and sort plugin state by key

File: src/components/search-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const searchBarWrapperStyles = css({
+  display: "flex",
+  gap: "4px",
+  margin: 4,
+});
+
+const SearchBar: React.FC<SearchBarProps> = ({ onSearch }) => {
+  const [query, setQuery] = useState("");
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (d4rkr00t) [inline src/components/search-bar.tsx]: should be useCallback

File: src/components/search-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    setQuery(event.target.value);
+    onSearch(event.target.value);
+  };
+
+  const handleSearch = () => {
```
Comment (d4rkr00t) [inline src/components/search-bar.tsx]: this one too :)

File: src/tabs/plugins.tsx [hunk 3]
```diff
@@ -19,40 +21,90 @@ export function PluginState(props: { pluginState: any }) {
...
   );
-  const state = useAtomValue(editorStateAtom);
-
-  if (!state) return null;
 
-  const plugins = state.plugins as any as Plugin<any>[];
-  const selectedPlugin = plugins[selectedIndex];
   const selectedPluginState = selectedPlugin.getState(state);
 
+  const handleSearch = (input: string) => {
```
Comment (d4rkr00t) [inline src/tabs/plugins.tsx]: and this one should be useCallback

**Review by d4rkr00t — COMMENTED** [review]
Also can you please prefix a commit with `feat: `? Then changelog will be generated automatically :)

**Pull request merged**